extensions/elytron-security-jdbc/runtime/src/main/java/io/quarkus/elytron/security/jdbc/JdbcRecorder.java

```diff
@@ -64,6 +64,7 @@ private void registerPrincipalQuery(PrincipalQueryConfig principalQuery, JdbcSec
 
     private DataSource getDataSource(PrincipalQueryConfig principalQuery) {
         return (DataSource) principalQuery.datasource
+                // this is not correct, it should be something like 'Arc.container().instance(javax.sql.DataSource.class, new DataSource.DataSourceLiteral(name))'
```

Comment:
Should we fix it?

Reply:
I wanted to make sure the tests run without the fix, because the way it's currently being used is good for uncovering bugs. 
I agree it should be fixed, in a second commit I guess?

Reply:
Yeah, I would say just add another commit. Better have it right!

Reply:
I added a second commit